### PR TITLE
Add the TestRunSwapMoreThanMemoryLimit case.

### PR DESCRIPTION
**Pull request by sunyuan3**
Signed-off-by: Yuan Sun sunyuan3@huawei.com

### CODE REVIEW

**Comment by sunyuan3** [comment]
It seems that jenkins can't work well.

File: integration-cli/docker_cli_run_unix_test.go [hunk 1]
```diff
@@ -338,3 +338,18 @@ func (s *DockerSuite) TestStopContainerSignal(c *check.C) {
 		c.Fatalf("Expected `exit trapped` in the log, got %v", out)
 	}
 }
+
+func (s *DockerSuite) TestRunSwapMoreThanMemoryLimit(c *check.C) {
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: this should be `TestRunSwapLessThanMemoryLimit`

Comment (sunyuan3) [inline integration-cli/docker_cli_run_unix_test.go]: Yes, you are right. I will update it.

File: integration-cli/docker_cli_run_unix_test.go [hunk 2]
```diff
@@ -338,3 +338,18 @@ func (s *DockerSuite) TestStopContainerSignal(c *check.C) {
...
+	testRequires(c, NativeExecDriver)
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: There's not need for this, lxc supports this and this tests won't even arrive to the execdriver

**Comment by sunyuan3** [comment]
@runcom , the code has been updated.

File: integration-cli/docker_cli_run_unix_test.go [hunk 3]
```diff
@@ -338,3 +338,17 @@ func (s *DockerSuite) TestStopContainerSignal(c *check.C) {
 		c.Fatalf("Expected `exit trapped` in the log, got %v", out)
 	}
 }
+
+func (s *DockerSuite) TestRunSwapLessThanMemoryLimit(c *check.C) {
+	testRequires(c, memoryLimitSupport)
+	testRequires(c, swapMemorySupport)
+	out, exitCode, err := dockerCmdWithError("run", "-m", "16m", "--memory-swap", "15m", "busybox", "echo", "test")
+
+	if err == nil || exitCode == 0 {
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: this if block can be replaced by just `c.Assert(err, check.NotNil)` and you can ignore `exitCode` in the command above (just `_` it)

File: integration-cli/docker_cli_run_unix_test.go [hunk 4]
```diff
@@ -338,3 +338,17 @@ func (s *DockerSuite) TestStopContainerSignal(c *check.C) {
...
+		c.Fatalf("Should not get right information: %s, %v", out, err)
+	}
+
+	if !strings.Contains(out, "Minimum memoryswap limit should be larger than memory limit") {
+		c.Fatal("couldn't find the invalid value warning")
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: something like `c.Fatalf("Expected %q, got %q", expected, out)` is better when tests fail and you want to see the output

**Comment by runcom** [comment]
@sunyuan3 tests failing

**Comment by tiborvass** [comment]
LGTM

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**